### ISSUE TITLE
ContentWriter bugged...

### ISSUE THREAD

**Issue by sharkist** [MonoGame/MonoGame#6661]
whenever I get the "ContentWriter.BaseStream" pointer, then the ContentWriter adding again and again the header in the file. Why ContentWrite write something in the file, i only get basestream.position?

If i do something like below:
```

        protected override void Write(ContentWriter cs_writer, TilesetContentData tsdata)
        {
            long a = cs_writer.BaseStream.Position;
            long b = cs_writer.BaseStream.Position;
            long c = cs_writer.BaseStream.Position;
            long d = cs_writer.BaseStream.Position;
            long e = cs_writer.BaseStream.Position;
            long f = cs_writer.BaseStream.Position;
            long g = cs_writer.BaseStream.Position;
        };

```
My file will have "8" XNB header files.

my workaround if i need 'basestream.position' just make first MemoryStream, and later everything copy by Write(memstream.ToArray());

[http://oi65.tinypic.com/2gx1eer.jpg](http://oi65.tinypic.com/2gx1eer.jpg)

#### What version of MonoGame does the bug occur on:
- MonoGame 3.7

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- DesktopGL

**Comment by Jjagg** [comment]
`ContentWriter` writes stuff when its `Flush` method is called. This is not correct behavior. Flush should not do the actual writing.

**Comment by h3allen** [comment]
Hi! I'm a first time contributor and I'd like to take a stab at fixing this if it's still a problem!

**Comment by mrhelmut** [comment]
I don't think this issue is still relevant. This behavior is inherited from ```BinaryWriter``` and doesn't seem to be a MonoGame issue. This should be tested against .NET 8 to be sure.

**Comment by Jjagg** [comment]
That doesn't sound right @mrhelmut. The `Flush` implementation should complete buffered write operations, but `ContentWriter` actually writes new data to the underlying stream.
You can see from the doc comment that the implementation has a wrong perception of what `Flush` should do:

https://github.com/MonoGame/MonoGame/blob/72919dc8185e70734507b58bd9a8e03bfe9a0d96/MonoGame.Framework.Content.Pipeline/Serialization/Compiler/ContentWriter.cs#L123

Compare with `BinaryWriter.Flush`'s [docs and implementation](https://github.com/dotnet/runtime/blob/589beb0b2af3edd1b272f8ac21ca5e3fb142bdd0/src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs#L123-L128):

```csharp
        // Clears all buffers for this writer and causes any buffered data to be
        // written to the underlying device.
        public virtual void Flush()
        {
            OutStream.Flush();
        }
```

The thing to figure out here is when we do need to write the stuff we now write in `Flush`.

**Comment by h3allen** [comment]
From the sounds of it, and from what I've seen, the current Flush call is working to finalize the file after the main body of content has been written, instead of only writing out the remaining data in the buffer. It seems as though this can either be slotted elsewhere in the ContentWriter class where it makes more sense or contained in its own method. What would be best given the current architecture?
